apps/chat/src/utils/app/application.ts

```diff
@@ -2,23 +2,37 @@ import {
   ApplicationInfo,
   CustomApplicationModel,
 } from '@/src/types/applications';
-import { EntityType } from '@/src/types/common';
+import { EntityType, PartialBy } from '@/src/types/common';
 import { DialAIEntityFeatures } from '@/src/types/models';
 
-import { getApplicationApiKey } from '../server/api';
+import { ApiUtils, getApplicationApiKey } from '../server/api';
 import { constructPath } from './file';
+import { getFolderIdFromEntityId } from './folders';
 import { getApplicationRootId } from './id';
 
 export const getGeneratedApplicationId = <T extends ApplicationInfo>(
-  application: Omit<T, 'folderId' | 'id'>,
+  application: Omit<T, 'id'>,
 ): string => {
   return constructPath(
     getApplicationRootId(),
     getApplicationApiKey(application),
   );
 };
 
-interface ApiApplicationModel {
+export const regenerateApplicationId = <T extends ApplicationInfo>(
```

Comment:
same here

Reply:
because in this case return type will be `ApplicationInfo`